File: dkb/format_dkb_test.go
package dkb

import (
	"rahul-rakshit/formatbankcsv/csv"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestFormatDkb_HappyPath(t *testing.T) {
	sampleDkbData, _ := csv.ReadCsv("./fixtures/sample_dkb.csv", ";", 4)

	formatted, _ := formatDkb(sampleDkbData)

	expectedResultCsv, _ := csv.ReadCsv("./fixtures/expected_result.csv", ",", 0)
	assert.Equal(t, expectedResultCsv, formatted)
}

func TestFormatN26_BadHeaderError(t *testing.T) {
	badHeaderData, _ := csv.ReadCsv("./fixtures/bad_header_dkb.csv", ";", 4)

	_, err := formatDkb(badHeaderData)

	assert.Equal(t, "Unexpected header for dkb format", err.Error())
}

func TestFormatN26_OldFormatError(t *testing.T) {
	_, err := csv.ReadCsv("./fixtures/old_dkb_format.csv", ";", 4)

	assert.Regexp(t, "record on line 3: wrong number of fields", err.Error())
}

File: dkb/formatDkb.test.ts
import { describe, it } from "@std/testing/bdd";
import { expect } from "@std/expect";
import { readCsv } from "../csv/csv.ts";
import { formatDkb } from "./formatDkb.ts";

describe("formatDkb", () => {
  it("formats dkb csv data correctly", async () => {
    const sampleDkbData = await readCsv("./dkb/fixtures/sampleDkb.csv", ";", 4);
    const formatted = formatDkb(sampleDkbData);
    const expectedResultCsv = await readCsv("./dkb/fixtures/expectedResult.csv", ",", 0);
    expect(formatted).toEqual(expectedResultCsv);
  });

  it("throws error for bad header", async () => {
    const badHeaderData = await readCsv("./dkb/fixtures/badHeaderDkb.csv", ";", 4);
    expect(() => formatDkb(badHeaderData)).toThrow("Unexpected header for dkb format");
  });

  it("throws error for old format", async () => {
    const oldFormatData = await readCsv("./dkb/fixtures/oldDkbFormat.csv", ";", 4);
    expect(() => formatDkb(oldFormatData)).toThrow("Unexpected header for dkb format");
  });
});
